src/WebJobs.Script/Binding/Http/HttpExtensions.cs

```diff
@@ -0,0 +1,284 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    public static class HttpExtensions
```

Comment:
Agree on internal but why not as extension methods? The serialize/deserialize methods are not specific to proxies and could be used outside of proxies as well.